pkg/genericapiserver/filters/maxinflight.go

```diff
@@ -18,33 +18,84 @@ package filters
 
 import (
 	"net/http"
+	"strings"
 
+	"k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/apiserver/request"
 	"k8s.io/kubernetes/pkg/httplog"
+	"k8s.io/kubernetes/pkg/util/sets"
+
+	"github.com/golang/glog"
 )
 
 // Constant for the retry-after interval on rate limiting.
 // TODO: maybe make this dynamic? or user-adjustable?
 const retryAfter = "1"
 
+var mutatingRequestVerbs = sets.NewString("PUT", "POST", "PATCH", "DELETE")
```

Comment:
'WATCH' is handled by the 'long running' part of the logic, so we can skip that. That's the reason why I wanted to have blacklist - it's a bit clearer for me.